### PR TITLE
TopK op

### PR DESCRIPTION
**Pull request by jamesr66a**
https://github.com/onnx/onnx/issues/492

### CODE REVIEW

File: onnx/backend/test/case/node/topk.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    @staticmethod
+    def export_top_k():
+        node = onnx.helper.make_node(
+            'TopK',
+            inputs=['x'],
+            outputs=['vales', 'indices'],
+            k=3
+        )
+        x = np.random.randn(3, 4, 5).astype(np.float32)
+        values, indices = top_k_ref(x, 3)
```
Comment (bddppq) [inline onnx/backend/test/case/node/topk.py]: The top_k_ref function code won't be extracted to the doc together (besides it's pretty long so might not be a good idea to put it into the doc either). Either put a hand-written simple inputs&outputs example here, or use np.sort with slice (you can select a stable sorting algorithm in np.sort).

File: onnx/defs/math/defs.cc [hunk 2]
```diff
@@ -496,3 +496,36 @@ OPERATOR_SCHEMA(MatMul)
...
+   the top k elements along the last dimension
+  -Index tensor of shape [a_1, a_2, ..., a_n, k] which contains the indices
+   of the top k elements (original indices from the input tensor).
+
+Given two equivalent values, this operator uses the indices along the last dim-
+ension as a tiebreaker. That is, the element with the lower index will appear
+first.
+)DOC")
+    .Input(0, "X", "Tensor of shape [a_1, a_2, ..., a_n, r]", "T")
+    .AllowConsumed({{2, 0}})
```
Comment (bddppq) [inline onnx/defs/math/defs.cc]: I think this means the Input[2] is allowed to be consumed by Output[0], and that's shouldn't be what you want :-)

**Comment by spandantiwari** [comment]
Would it make sense to introduce an ``axis`` attribute to find the opt k elements along a user-specified dimension? Default value can be the last axis (-1). For instance, Numpy's ``argsort``, which is commonly used for getting top k elements, allows this as an input argument.

**Comment by jamesr66a** [comment]
@spandantiwari Yep, good suggestion. Updated

**Comment by jamesr66a** [comment]
@onnxbot retest this please

File: docs/Operators.md [hunk 3]
```diff
@@ -6253,6 +6254,98 @@ opset_import {
...
+~~~~
+opset_import {
+  version = 1
+}
+~~~~
+
+#### Attributes
+
+<dl>
+<dt><tt>dim</tt> : int</dt>
```
Comment (spandantiwari) [inline docs/Operators.md]: My recommendation would be to name this attribute ``axis``, in line with other ONNX ops such as ``concat``, ``div``, ``flatten``, ``gather`` and others.

**Pull request merged**